### PR TITLE
Can restore from any checkpoint, not just latest

### LINKED ISSUE #768: Save/restore for TensorGraph
In the past, our models have generally had a persistent Session object containing all the variables for the model.  They automatically wrote out checkpoints during training, and provided a `restore()` method to load the variables from the most recent checkpoint into memory.

TensorGraph works differently.  It creates a new Session every time you call `fit()` or `predict()`, loads the variables from the latest checkpoint, then throws away the Session before returning.  I find this behavior has a couple of problems.

First, it's really slow.  If you want to run prediction thousands of times, that means it will load the variables from disk thousands of times.  This is why for both RL and MAML I've basically had to subvert the design of TensorGraph.  I use it to build the Tensorflow graph, but when I need to do any calculations I ignore the corresponding parts of TensorGraph, pull out the needed internal fields, and do the calculations directly.

It also is inflexible, since it enforces that prediction can only ever be done based on the variables in the latest checkpoint file, no others.  A good example of where this causes problems is MAML.  It uses TensorGraph to define the model, but the actual optimization is done with a different optimizer and a different loss function.  That produces a model (which gets saved to disk) that is designed to be easy to train, but isn't optimized for any particular task.  To use it for prediction, you first do a few steps of gradient descent to produce a fine tuned version of the model (which does *not* get saved to disk), and do the prediction based on that.  But I can't use any of TensorGraph's prediction methods for it, because it would just throw away the tuned variables and replace them with the generic ones from disk.

I think we should consider changing this behavior.

### PR DESCRIPTION
**Pull request by peastman**
Fixes #768.

### CODE REVIEW

File: deepchem/models/tensorgraph/tensor_graph.py [hunk 1]
```diff
@@ -765,16 +765,30 @@ def save_checkpoint(self, max_checkpoints_to_keep=5):
...
     saver.save(self.session, self.save_file, global_step=self.global_step)
 
-  def restore(self):
-    """Reload the values of all variables from the most recent checkpoint file."""
+  def get_checkpoints(self):
+    """Get a list of all available checkpoint files."""
+    return tf.train.get_checkpoint_state(
+        self.model_dir).all_model_checkpoint_paths
+
+  def restore(self, checkpoint):
```
Comment (rbharath) [inline deepchem/models/tensorgraph/tensor_graph.py]: Should `checkpoint` be a kwarg `checkpoint=None`?

Comment (peastman) [reply deepchem/models/tensorgraph/tensor_graph.py]: Yes, thanks.  I just fixed it.

**Comment by rbharath** [comment]
LGTM
